### PR TITLE
[Feature]support tp for deepseek checkpoint conversion

### PR DESCRIPTION
**Pull request by lxd-cumt**
support tp for deepseek checkpoint conversion

### CODE REVIEW

File: tools/checkpoint/deepseek_v3/ckpt.py [hunk 1]
```diff
@@ -459,74 +496,82 @@ def get_moe_mlp_ckpt(message, models, layer_id, args):
...
-                    expert_down_weight = expert.lienar_fc2.weight.data
+                    expert_linear1_weight.append(expert.linear_fc1.weight.data)
+                    expert_linear2_weight.append(expert.linear_fc2.weight.data)
                 else:  # using TEGroupedMLP
-                    expert_linear_fc1_weight = getattr(
-                        tf_layer.mlp.experts.linear_fc1, f"weight{expert_id}", None
+                    expert_linear1_weight.append(
+                        getattr(
+                            tf_layer.mlp.experts.linear_fc1, f"weight{expert_id}", None
+                        ).detach()
```
Comment (zhaoyinglia) [inline tools/checkpoint/deepseek_v3/ckpt.py:515]: why detach?

Comment (lxd-cumt) [reply tools/checkpoint/deepseek_v3/ckpt.py:515]: The function **getattr()** return a non-leaf tensor with grad func, but is does support corss-process boundaries.
Therefore, if we want to transfer data by using multi-process queue in ckpt conversion, we need to call detach() on the tensor before serializing.

**Review by zhaoyinglia — APPROVED** [review]
LGTM
